docs/proposals/volumes.md

```diff
@@ -0,0 +1,373 @@
+not be appropriate to perform `chown` or `chmod` operations on some distributed file systems.
+For example, actions taken by root on an NFS client are treated as `nobody` on the server unless the
+`no_squash_root` setting is enabled for that volume's NFS export.  Enabling `no_squash_root`
+contradicts Red Hat's security guidance for NFS (as an example), so there will need to be another
+mechanism that prepares some distributed fs volumes.
+
+One possibility is that if the server hosting the filesystem is itself a Kubernetes node, a pod
+could be scheduled onto that node that prepared the volume by setting the ownership, permissions,
+and SELinux context.
+
+### Relabeling volumes with correct SELinux context
+
+On systems with SELinux enabled, volumes should be relabeled with the correct SELinux context.
+Docker has this capability today; it is desireable for other container runtime implementations to
+provide similar functionality.
+
+Relabeling should be an optional aspect of a volume plugin to accomodate:
+
+1.  volume types for which generalized relabeling support is not sufficient
+2.  testing for each volume plugin individually
+
+#### SELinux and distributed file systems
+
+Some distributed filesystems have complications where SELinux is involved.  Let's look at NFS as an
+example.  Files in NFS volumes cannot have their contexts changed with `chcon` after the volume is
+mounted; the `context` argument must be passed to the `mount` call.
+
+For NFS, there is an SELinux boolean, `virt_use_nfs` that allows containers to use files with the
+SELinux type `nfs_t`.  Setting `virt_use_nfs` on a system would allow containers to use *any* NFS
+mount without the kubelet having to relabel the files.  However, the semantics would be different
+from what might be expected, since SELinux will not provide any protection against containers from
+one pod using files in an NFS mount belonging to another pod.
+
+NFS is a case where the `chcon` approach isn't going to solve the whole problem.  Instead, the NFS
+volume plugin will have to handle calling `mount` with the `context` argument in order to gain
+cross-container protection from SELinux.
+
+Not all distributed file systems support SELinux uniformly.  Some file systems provide only the
+coarse-level control via an SELinux boolean to determine whether containers can use them.  The
+internal API should account for this.
+
+## Proposed Design
+
+Our proposed design should minimize code for handling ownership and SELinux required in:
+
+1.  volume plugins
+2.  the kubelet
+
+### API changes
+
+There are a couple factors to consider for the API:
+
+1.  Currently, it is possible to specify a different SELinux context for each container in a pod
+2.  There must be a way to specify a supplemental group that all containers in a pod are part of
+3.  There should be a uniform way to specify in the API that a volume requires ownership / label
+    management
```

Comment:
You should state and link to your pod level security proposal here. Also, I am not sure if that proposal explicitly states that volumes will inherit the security context of the pod. If not I think you should mentioned that